dask_kubernetes/core.py

```diff
@@ -317,226 +502,41 @@ def namespace(self):
-
-    def scale_up(self, n, pods=None, **kwargs):
-        """
-        Make sure we have n dask-workers available for this cluster
-
-        Examples
-        --------
-        >>> cluster.scale_up(20)  # ask for twenty workers
-        """
-        maximum = dask.config.get("kubernetes.count.max")
-        if maximum is not None and maximum < n:
-            logger.info(
-                "Tried to scale beyond maximum number of workers %d > %d", n, maximum
-            )
-            n = maximum
-        pods = pods or self._cleanup_terminated_pods(self.pods())
-        to_create = n - len(pods)
-        new_pods = []
-        for i in range(3):
-            try:
-                for _ in range(to_create):
-                    new_pods.append(
-                        self.core_api.create_namespaced_pod(
-                            self.namespace, self.pod_template
-                        )
-                    )
-                    to_create -= 1
-                break
-            except kubernetes.client.rest.ApiException as e:
-                if e.status == 500 and "ServerTimeout" in e.body:
-                    logger.info("Server timeout, retry #%d", i + 1)
-                    time.sleep(1)
-                    last_exception = e
-                    continue
-                else:
-                    raise
-        else:
-            raise last_exception
-
-    def scale_down(self, workers, pods=None):
-        """ Remove the pods for the requested list of workers
-
-        When scale_down is called by the _adapt async loop, the workers are
-        assumed to have been cleanly closed first and in-memory data has been
-        migrated to the remaining workers.
-
-        Note that when the worker process exits, Kubernetes leaves the pods in
-        a 'Succeeded' state that we collect here.
-
-        If some workers have not been closed, we just delete the pods with
-        matching ip addresses.
-
-        Parameters
-        ----------
-        workers: List[str] List of addresses of workers to close
+        logs: Dict[str]
+            A dictionary of logs, with one item for the scheduler and one for
+            each worker
         """
-        # Get the existing worker pods
-        pods = pods or self._cleanup_terminated_pods(self.pods())
+        logs = Logs()
 
-        # Work out the list of pods that we are going to delete
-        # Each worker to delete is given in the form "tcp://<worker ip>:<port>"
-        # Convert this to a set of IPs
-        ips = set(urlparse(worker).hostname for worker in workers)
-        to_delete = [p for p in pods if p.status.pod_ip in ips]
-        if not to_delete:
-            return
-        self._delete_pods(to_delete)
+        if scheduler:
+            logs["Scheduler"] = await self.scheduler.logs()
 
-    def __enter__(self):
-        return self
+        if workers:
+            for key, worker in self.workers.items():
+                logs[key] = await worker.logs()
```

Comment:
I suspect that we will want to await many of these at once, maybe something like this:

```python
worker_logs = await asyncio.gather(*[w.logs() for w in self.workers.values()])
for key, log in zip(self.workers, worker_logs):
    logs[key] = log
```